control/grpc.py

```diff
@@ -2452,15 +2465,11 @@ def get_subsystems_safe(self, request, context):
                 ns_key = "namespaces"
                 ns_list = []
                 if ns_key in s:
-                    ns_list = s[ns_key]
-                if not ns_list:
-                    self.subsystem_nsid_bdev_and_uuid.remove_namespace(nqn)
```

Comment:
I think we should keep this. It costs nothing and we have a lot of bugs in the processing of local lists. This is a safeguard against bugs in removing deleted namespaces.

Reply:
I guess now, such an operation adds the need of taking the shared lock. I guess we can remove it but we need to make sure we didn't miss anything in maintaining this list.